// sysctrl/sysctrl.js
'use strict'
var sysctrl = require('bindings')('sysctrl.node');

function ctrl(info, callback) {
	sysctrl.netcamconn(info, callback);
}

ctrl.prototype.PTZ_CMD_STOP = 0;			//停止
ctrl.prototype.PTZ_CMD_LEFT = 1;			//左
ctrl.prototype.PTZ_CMD_RIGHT = 2;			//右
ctrl.prototype.PTZ_CMD_UP = 3;				//上
ctrl.prototype.PTZ_CMD_DOWN = 4;			//下
ctrl.prototype.PTZ_CMD_LEFT_UP = 5;			//左上
ctrl.prototype.PTZ_CMD_LEFT_DOWN = 6;		//左下
ctrl.prototype.PTZ_CMD_RIGHT_UP = 7;		//右上
ctrl.prototype.PTZ_CMD_RIGHT_DOWN = 8;		//右下
ctrl.prototype.PTZ_CMD_NEAR = 9;			//拉近
ctrl.prototype.PTZ_CMD_FAR = 10;			//调远
ctrl.prototype.PTZ_CMD_ZOOM_OUT = 11;		//缩小
ctrl.prototype.PTZ_CMD_ZOOM_IN = 12;		//放大
ctrl.prototype.PTZ_CMD_IRIS_OPEN = 13;		//开启光圈
ctrl.prototype.PTZ_CMD_IRIS_CLOSE = 14;		//关闭光圈

ctrl.prototype.netcamctrl = function(playHandle, flag) {
	sysctrl.netcamctrl(playHandle, flag);
};

ctrl.prototype.netcamclose = function(userId, playHandle) {
	if(playHandle > 0) {
		sysctrl.netcamctrl(playHandle, ctrl.prototype.PTZ_CMD_STOP);
	}
	sysctrl.netcamclose(userId, playHandle);
};

module.exports = ctrl;
